### ISSUE TITLE
Multiworksheet Change Propagation

### ISSUE THREAD

**Issue by kousu** [audreyt/ethercalc#576]
I have set up a spreadsheet (on ethercalc@0.20170704.0, just updated from npm today) with three worksheets in it. The third worksheet has references to the second which reference the first, and direct references to the first.

Here's my first sheet:

![screenshot from 2018-01-02 03-57-16](https://user-images.githubusercontent.com/987487/34478837-07a2e290-ef9b-11e7-9f26-7a22a4b02de1.png)

the intermediate sheet:

![screenshot from 2018-01-02 03-58-53](https://user-images.githubusercontent.com/987487/34478856-28c82ebc-ef9b-11e7-984e-e13eb94ef314.png)

and the output sheet:

![screenshot from 2018-01-02 03-52-08](https://user-images.githubusercontent.com/987487/34478874-4238d5ea-ef9b-11e7-8a0e-ca671d256fe9.png)

If I change the first sheet by adding a new data row:

![screenshot from 2018-01-02 03-53-18](https://user-images.githubusercontent.com/987487/34478898-6158a450-ef9b-11e7-89e8-4f4c54a1dd5b.png)

I get this error, where my sums don't add up:

![screenshot from 2018-01-02 03-52-56](https://user-images.githubusercontent.com/987487/34478902-65dbef1e-ef9b-11e7-9691-42d30b4d423d.png)

If you trace the numbers, the cause of this is that is that the second row has updated while the first row hasn't.

The first row has these formulas:

```
=SUM("bud2018feb.2"!B1:B99)
```

which make totals out of the intermediate sheet's

```
=IF(NOT(ISBLANK("bud2018feb.1"!$A5)),"bud2018feb.1"!$D5*(1/COLUMNS($B$1:$G$1)),"")
# this basically says
="bud2018feb.1"!$D5/6
# just with some error checking wrapped around
```

and the second row has

```
=SUMIF("bud2018feb.1"!$B$1:$B$100, B1, "bud2018feb.1"!$D$1:$D$100)
```

So the first row of sheet 3 references sheet 2 which references sheet 1, and the second row of sheet 3 references sheet 1 directly. And changing sheet 1 updated the second but not first row.

Plus, I tried tweaking one of the entries in sheet 2 and that seems to have triggered the rest of the recompute of sheet 3.

Here is the full example; it's in SocialCalc-save format in text files in a zip, so you'll have to manually paste it together again, but I think it should survive the transition.

[bud2018feb.zip](https://github.com/audreyt/ethercalc/files/1597108/bud2018feb.zip)

**Comment by kousu** [comment]
By the way, I'm super impressed at how much of Spreadsheetlandia you have working. There's glitches, like this, but by and large Ethercalc is usable for real tasks already. That's pretty cool.

**Comment by eddyparkinson** [comment]
Recompute
---
Correct on load/reload - To get the correct calculations, suggest you reload. If you are updating several sheets, suggest recalculate by reloading. Because when a sheet loads or reloads, it should produce the right values, as the values on the server should be correct. But only the visible sheet is kept fresh, other sheets are not updated in the browser when changes happen.

**Comment by kousu** [comment]
So is that by design, to save CPU?

What about this compromise: recompute an entire worksheet when its tab is switched to, but not before.

**Comment by eddyparkinson** [comment]
>So is that by design, to save CPU?

no. Just history. This could be added. Would be handy. Suspect it would require a few days of work to change how this works.
